verifiers/internal/gha/slsaprovenance/v1.0/byob.go

```diff
@@ -2,239 +2,92 @@ package v1
+		}
+		return common.GetBranch(sysParams, true)
 	}
 
-	workflowRef, err := common.GetAsString(sysParams, "GITHUB_WORKFLOW_REF")
+	// Returns the branch from the source URI if available.
+	_, ref, err := utils.ParseGitURIAndRef(sourceURI)
 	if err != nil {
-		return "", "", "", err
+		return "", fmt.Errorf("parsing source uri: %w", err)
 	}
 
-	parts := strings.Split(workflowRef, "@")
-	if len(parts) != 2 {
-		return "", "", "", fmt.Errorf("%w: ref: %s", serrors.ErrorInvalidFormat, workflowRef)
+	if ref == "" {
+		return "", fmt.Errorf("%w: unable to get ref for source %q",
+			serrors.ErrorInvalidDssePayload, sourceURI)
 	}
-	repoAndPath := parts[0]
-	ref := parts[1]
 
-	parts = strings.Split(repoAndPath, "/")
-	if len(parts) < 2 {
-		return "", "", "", fmt.Errorf("%w: rep and path: %s", serrors.ErrorInvalidFormat, repoAndPath)
+	refType, _ := utils.ParseGitRef(ref)
+	switch refType {
+	case "heads": // branch.
+		// NOTE: We return the full git ref.
+		return ref, nil
+	case "tags":
+		// NOTE: If the ref type is a tag we want to try to parse out the branch from the tag.
+		sysParams, ok := p.prov.Predicate.BuildDefinition.InternalParameters.(map[string]interface{})
+		if !ok {
+			return "", fmt.Errorf("%w: %s", serrors.ErrorInvalidDssePayload, "internal parameters type")
+		}
+		return common.GetBranch(sysParams, true)
+	default:
+		return "", fmt.Errorf("%w: unknown ref type %q for ref %q",
+			serrors.ErrorInvalidDssePayload, refType, ref)
 	}
-
-	repo := strings.Join(parts[:2], "/")
-	path := strings.Join(parts[2:], "/")
-	return fmt.Sprintf("git+https://github.com/%s", repo), ref, path, nil
-}
-
-func (p *BYOBProvenance) triggerInfo() (string, string, string, error) {
-	// TODO(#613): Support for generators.
-	return p.builderTriggerInfo()
-}
-
-// TriggerURI implements Provenance.TriggerURI.
-func (p *BYOBProvenance) TriggerURI() (string, error) {
-	repository, ref, _, err := p.triggerInfo()
-	if err != nil {
-		return "", err
-	}
-	if repository == "" || ref == "" {
-		return "", fmt.Errorf("%w: repository or ref is empty", serrors.ErrorMalformedURI)
-	}
-	return fmt.Sprintf("%s@%s", repository, ref), nil
-}
-
-// Subjects implements Provenance.Subjects.
-func (p *BYOBProvenance) Subjects() ([]intoto.Subject, error) {
-	subj := p.prov.Subject
-	if len(subj) == 0 {
-		return nil, fmt.Errorf("%w: %s", serrors.ErrorInvalidDssePayload, "no subjects")
-	}
-	return subj, nil
-}
-
-// GetBranch implements Provenance.GetBranch.
-func (p *BYOBProvenance) GetBranch() (string, error) {
-	// TODO(https://github.com/slsa-framework/slsa-verifier/issues/472): Add GetBranch() support.
-	sysParams, ok := p.prov.Predicate.BuildDefinition.InternalParameters.(map[string]interface{})
-	if !ok {
-		return "", fmt.Errorf("%w: %s", serrors.ErrorInvalidDssePayload, "internal parameters type")
-	}
-
-	return common.GetBranch(sysParams, true)
 }
 
 // GetTag implements Provenance.GetTag.
 func (p *BYOBProvenance) GetTag() (string, error) {
```

Comment:
where do we check whether sha1 != GH event sha1? Maybe in a follow-up PR?

Reply:
We don't actually do any sha checks since we don't verify the specific sha against a user provided value yet. The only change to make really was to fail if the user expects a specific branch because when sha1 != GH sha we can't actually determine that.

Reply:
gotcha. Thanks!